### PR TITLE
Ticket #1015 Update availability API to use EPP availability check

### LINKED ISSUE #1015: Update /availability API to pull from new registry
### Issue Description

The [current API](https://github.com/cisagov/getgov/blob/784cc0f618e056c262512d688e8e4316dd25c9e4/src/api/views.py#L14) consists of a second-hand pull of canonical data that the .gov program [publishes to GitHub](https://github.com/cisagov/dotgov-data/blob/main/current-full.csv). Change this implementation so that queries poll the new registry/Whois/RDAP.

### Acceptance Criteria

- [x] The checkDomain method is used to check that a domain is in the registry or not

and

- [x] The check is used at /availability 
- [x] Tests are implemented and/or updated
- [ ]  In the new domain application flow, where a user is checking if a domain is available for use, it should now use this /availability endpoint (if not already) and tests should be updated to check that /availability and epp is being triggered on this page as well. 

### Additional Context (optional)

Once done, the backend work of #476 and frontend work of #561 can be completed.

_Consider add-on to defend the endpoint from bot spam._

### Implementation Notes

Epp has a check command that can be run with just the name of a given domain.  This should be used as the method for checking the desired domain at this endpoint.

### Issue Links

Blocking #476 and #561
Blocked by #1028

### PR DESCRIPTION
**Pull request by erinysong**
## Ticket

Resolves #1015 

## Changes

- Update availability API to replace current availability check via comparing public GitHub file to using the EPP availability check function CheckDomain
- Update availability API and form tests to mock EPP responses of expected test outcomes.

## Setup

**Test locally**
Run tests to confirm that updated availability API works.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/api/views.py [hunk 1]
```diff
@@ -59,12 +59,12 @@ def in_domains(domain):
     given domain doesn't end with .gov, ".gov" is added when looking for
     a match.
     """
-    domain = domain.lower()
+    Domain = apps.get_model("registrar.Domain")
     if domain.endswith(".gov"):
-        return domain.lower() in _domains()
+        return Domain.available(domain)
```
Comment (therealslimhsiehdy) [inline src/api/views.py:64]: **(Very Dumb Question):** I'm trying to figure out where exactly it does the check for case sensitivity...would you be able to point me to where?

From the `available` function in getgov -- I see `string_could_be_domain` for a regex, but when I go from `Domain.available` -> `commands.CheckDomain` (found in the `epplib` folder) -> ` _get_check_payload`

I'm assuming it'd be done in `_get_check_payload`, but I'm trying to find the logic that fully convinces me of the full check 😅

**Review by therealslimhsiehdy — APPROVED** [review]
I was able to run the tests and works smoothly -- I had one non-blocking question, but other than that looks great 💯

File: src/api/views.py (hunk 1 above)
Comment (erinysong) [reply src/api/views.py:64]: Yes of course! Rachid made a series of tests in [test_models_domain.py](https://github.com/cisagov/getgov/blob/5c0e19397449d87815c70ebf2b896ffe66852945/src/registrar/tests/test_models_domain.py#L398) that verify that Domain.available -> commands.CheckDomain handle case sensitivity in domains - if you think it'd be good to add that case sensitivity in our API check I can also add that in in another PR!

Comment (therealslimhsiehdy) [reply src/api/views.py:64]: It's all good! I saw that you handled it in your PR tests (which is great!), was just trying to wrap my head around the logic of how CheckDomain in EPPLib did it -- EPPLib truly is magical 😆 Thanks for confirming!!

**Pull request merged**